File: Java/Facade/MovieStreamFacade/src/subsystem/StreamingDevice.java
package subsystem;

public class StreamingDevice {
    public void connect() {
        System.out.println("connect to stream device");
    }

    public void disconnect() {
        System.out.println("disconnect from stream device");
    }

    public void play(String movie) {
        System.out.println("playing " + movie);
    }
}

File: Golang/Facade/Beginner/subsystems/streaming_device.go
package subsystems

import (
	"fmt"
)

type StreamingDevice struct {}

func (s *StreamingDevice) Connect() {
	fmt.Println("Streaming device connected")
}

func (s *StreamingDevice) Disconnect() {
	fmt.Println("Streaming device disconnected")
}

func (s *StreamingDevice) Play(movie string) {
	fmt.Printf("Playing movie %s\n", movie)
}